hack/scripts/kernel.sh

```diff
@@ -0,0 +1,123 @@
+    make -C ../aks ${CLUSTER_TYPE} \
+    AZCLI=az REGION=${REGION} SUB=${SUB} \
+    CLUSTER=${CLUSTER_NAME} \
+    VM_SIZE=${VMSIZE} \
+    AUTOUPGRADE=none
+else
+    echo Cluster is not being created through script. Proceeding with upgrade.
+fi
+
+if [ $DUMMY_CNI = "true" ]; then
+    echo "-- Install dummy CNI for nodes to be marked as Ready --"
+    kubectl get pods -Aowide
+    kubectl apply -f https://raw.githubusercontent.com/Azure/azure-container-networking/v1.5.3/hack/manifests/cni-installer-v1.yaml
+    kubectl rollout status ds -n kube-system azure-cni
+fi
+
+echo "-- Start privileged daemonset --"
+kubectl get pods -Aowide
+kubectl apply -f ../../test/integration/manifests/load/privileged-daemonset.yaml
+sleep 10s
+kubectl rollout status ds -n kube-system privileged-daemonset
+
+echo "-- Update kernel through daemonset --"
+kubectl get pods -n kube-system -l os=linux,app=privileged-daemonset -owide
+privList=`kubectl get pods -n kube-system -l os=linux,app=privileged-daemonset -owide --no-headers | awk '{print $1}'`
+for pod in $privList; do
+    echo "-- Update Ubuntu Packages --"
+    # Not needed, but ensures that the correct packages exist to perform upgrade
+    kubectl exec -i -n kube-system $pod -- bash -c "apt update && apt-get install software-properties-common -y"
+
+    echo "-- Add proposed repository --"
+    kubectl exec -i -n kube-system $pod -- bash -c "add-apt-repository ppa:canonical-kernel-team/proposed -y"
+    kubectl exec -i -n kube-system $pod -- bash -c "add-apt-repository ppa:canonical-kernel-team/proposed2 -y"
+
+    echo "-- Check apt-cache --"
+    kubectl exec -i -n kube-system $pod -- bash -c "apt-cache madison linux-azure-edge"
+
+    echo "-- Check current Ubuntu kernel --"
+    kubectl exec -i -n kube-system $pod -- bash -c "uname -r"
+    kubectl get node -owide
+
+    echo "-- Install Proposed Kernel --"
+    kubectl exec -i -n kube-system $pod -- bash -c "apt install -y linux-azure-edge"
+done
+
+privArray=(`kubectl get pods -n kube-system -l os=linux,app=privileged-daemonset -owide --no-headers | awk '{print $1}'`)
+nodeArray=(`kubectl get pods -n kube-system -l os=linux,app=privileged-daemonset -owide --no-headers | awk '{print $7}'`)
+kubectl get pods -n kube-system -l os=linux,app=privileged-daemonset -owide
+
+i=0
+for _ in ${privArray[@]}; do
+    echo "-- Restarting Node ${nodeArray[i]} through ${privArray[i]} --"
+    kubectl exec -i -n kube-system ${privArray[i]} -- bash -c "reboot"
+    echo "-- Waiting for condition NotReady --"
+    kubectl wait --for=condition=Ready=false -n kube-system pod/${privArray[i]} --timeout=90s
+    echo "-- Waiting for condition Ready --"
+    kubectl wait --for=condition=Ready -n kube-system pod/${privArray[i]} --timeout=90s
+    ((i = i + 1))
+    echo "Wait 10s for pods to settle"
+    sleep 10s
+done
+
+kubectl rollout status ds -n kube-system privileged-daemonset
+for pod in $privList; do
+    echo "-- Check current Ubuntu kernel --"
```

Comment:
This should say upgraded Ubuntu kernel ? Or maybe check with the earlier version and echo out if the upgrade happened or not.

Reply:
I think the current string makes sense as we are not doing anything within bash to confirm that it is in fact upgraded. This script is meant to be used manually and we would be visually confirming the kernel upgraded as well.